lib/ansible/modules/monitoring/zabbix/zabbix_proxy.py

```diff
@@ -165,6 +172,16 @@ def add_proxy(self, data):
                 if data[item]:
                     parameters[item] = data[item]
 
+            if 'proxy_address' in parameters:
+                parameters.pop('proxy_address')
+            if 'proxy_address' in data and data['status'] == '5':
+                parameters['proxy_address'] = data['proxy_address']
+
+            if 'interface' in parameters:
+                parameters.pop('interface')
+            if 'interface' in data and data['status'] == '6':
+                parameters['interface'] = data['interface']
+
```

Comment:
Is there a reason to handling this differently than changes you've introduced in `update_proxy()` at lines 228-232?

Reply:
yes - if handled the same way ansible result was always "changed". This way it works properly if nothing is changed.

Reply:
Not required to change, but something like this seems to be working for me in the same way:

```python
            if 'proxy_address' in data and data['status'] != '5':
                parameters.pop('proxy_address', False)

            if 'interface' in data and data['status'] != '6':
                parameters.pop('interface', False)

```